lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -414,7 +414,7 @@ LowererMDArch::LoadFuncExpression(IR::Instr *instrFuncExpr)
         paramOpnd = IR::SymOpnd::New(paramSym, TyMachReg, this->m_func);
     }
 
-    if (this->m_func->GetJnFunction()->IsGenerator())
+    if (instrFuncExpr->m_func->GetJnFunction()->IsGenerator())
```

Comment:
Yes and no. It’s not directly related to the feature, but there are a few things that fix bugs the feature exposed in the UT’s, mainly by enabling more optimization.

--Paul

From: Ian Halliday [mailto:notifications@github.com]
Sent: Monday, June 27, 2016 2:48 PM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Paul Leathers <pleath@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Microsoft/ChakraCore] Precise closure-capture of local variables in the presence of deferred nested functions. This is a good thing to do anyway (improves CQ and solves some odd memory leak cases) and is especially important for redeferral and it ...


In lib/Backend/amd64/LowererMDArch.cpp<https://github.com/Microsoft/ChakraCore/pull/1167#discussion_r68662568>:

> @@ -414,7 +414,7 @@ LowererMDArch::LoadFuncExpression(IR::Instr *instrFuncExpr)

>          paramOpnd = IR::SymOpnd::New(paramSym, TyMachReg, this->m_func);

>      }

>

> -    if (this->m_func->GetJnFunction()->IsGenerator())

> +    if (instrFuncExpr->m_func->GetJnFunction()->IsGenerator())

Is this a drive by bug fix?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/1167/files/db13fe85a2a4226e5774ba57f86e893841f00c78#r68662568>, or mute the thread<https://github.com/notifications/unsubscribe/APF8RD2bUzCmcGt7QtBZ9wOqU4wROmaQks5qQEUWgaJpZM4I7R1X>.